Telegram/SourceFiles/history/history_message.cpp

```diff
@@ -1142,9 +1142,9 @@ bool HistoryMessage::drawBubble() const {
 }
 
 QRect HistoryMessage::countGeometry() const {
-	auto maxwidth = qMin(st::msgMaxWidth, _maxw);
+	auto maxwidth = std::min(st::msgMaxWidth, _maxw);
 	if (_media && _media->currentWidth() < maxwidth) {
-		maxwidth = qMax(_media->currentWidth(), qMin(maxwidth, plainMaxWidth()));
+		maxwidth = std::max(_media->currentWidth(), std::min(maxwidth, plainMaxWidth()));
```

Comment:
clamp

Reply:
Clamp cant be used here. There may be situations it will lead to undefined behavior, this constuction doesnt guarantee that second argument of std:max greater 0 and first argument of std::min greater 0, or one of these greater than other

From specs of std::clamp
"The behavior is undefined if the value of lo is greater than hi"